### PR TITLE
Fixes #18538: Better default value for dns_reverse

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
So the tests are failing with a weird error and I can't figure out why. Putting it here so others can have a look at it as well.

File: manifests/proxydns.pp [hunk 1]
```diff
@@ -9,19 +9,32 @@
...
+  }
+
+  ::dns::zone { $::foreman_proxy::dns_zone:
     soa     => $::fqdn,
     reverse => false,
     soaip   => $ip,
   }
 
-  ::dns::zone { $foreman_proxy::dns_reverse:
+  ::dns::zone { $reverse:
```
Comment (stbenjam) [inline manifests/proxydns.pp]: This error?

Evaluation Error: Missing title. The title expression resulted in undef at /home/travis/build/theforeman/puppet-foreman_proxy/spec/fixtures/modules/foreman_proxy/manifests/proxydns.pp:37:17 on node testing-docker-24f9b60e-1948-419b-bfa3-add6f0487c67.ec2.internal

Maybe the if check above on  line 26 isn't working correctly, and reverse is getting passed as undef?

Comment (ekohl) [inline manifests/proxydns.pp]: I found it: the parser function needs ```:type => :rvalue```.

**Comment by ekohl** [comment]
Updated with additional checks. I've also added some improvements to proxydns while I was at it, but I can split those into a separate commit if needed.

**Comment by mmoll** [comment]
I didn't test this, but the tests look meaningfully good to me.

**Comment by ekohl** [comment]
Removed a redundant ```.to_s```. And I didn't test it either other than the spec tests.

I should also say that I took a bit of a shortcut. You could also return an array so a /22 is multiple /24 blocks. I honestly don't know what people would prefer though IMHO if you're serious about your network and have a complex structure you won't use this class anyway.

File: manifests/params.pp [hunk 2]
```diff
@@ -290,7 +290,7 @@
   $dns_interface          = 'eth0'
   $dns_zone               = $::domain
   $dns_realm              = upcase($dns_zone)
-  $dns_reverse            = '100.168.192.in-addr.arpa'
+  $dns_reverse            = undef
```
Comment (domcleal) [inline manifests/params.pp]: Also update the documented `Variant` data type to add `Undef` as a permitted value.

Comment (ekohl) [inline manifests/params.pp]: Updated

**Comment by domcleal** [comment]
Merged, thanks @ekohl.

**Pull request merged**